decryptor/mysql/prepared_statement_sql_observer.go

```diff
@@ -0,0 +1,519 @@
+			continue
+		}
+
+		tableName := strings.TrimPrefix(splits[0], "@")
+		columnName := splits[1]
+
+		schema := e.schemaStore.GetTableSchema(tableName)
+		if schema == nil {
+			logrus.Debugf("Hasn't schema for table in SET query %s", tableName)
+			return nil, false, nil
+		}
+
+		columnSetting := schema.GetColumnEncryptionSettings(columnName)
+		if columnSetting == nil {
+			logrus.Debugf("No column encryption setting %s", columnName)
+			return nil, false, nil
+		}
+
+		accessContext := base.AccessContextFromContext(ctx)
+		clientID := columnSetting.ClientID()
+		if len(clientID) > 0 {
+			logrus.WithField("client_id", string(clientID)).Debugln("Encrypt with specific ClientID for column")
+		} else {
+			logrus.WithField("client_id", string(accessContext.GetClientID())).Debugln("Encrypt with ClientID from connection")
+			clientID = accessContext.GetClientID()
+		}
+
+		rawData, err := e.coder.Decode(sqlVal, columnSetting)
+		if err != nil {
+			return nil, false, nil
+		}
+
+		encryptedData, err := e.encryptor.EncryptWithClientID(clientID, rawData, columnSetting)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if !bytes.Equal(encryptedData, sqlVal.Val) {
+			output := make([]byte, hex.EncodedLen(len(encryptedData)))
+			hex.Encode(output, encryptedData)
+
+			sqlVal.Val = output
+			sqlVal.Type = sqlparser.HexVal
+			changedRes = true
+		}
+	}
+
+	return base.NewOnQueryObjectFromStatement(setQuery, e.parser), changedRes, nil
+}
+
+func (e *PreparedStatementsQuery) onExecute(ctx context.Context, executeQuery *sqlparser.Execute) (base.OnQueryObject, bool, error) {
+	logrus.Debugln("PreparedStatementsQuery.Execute")
+
+	var preparedStatementName = executeQuery.PreparedStatementName.String()
+
+	_, err := e.proxyHandler.registry.StatementByID(preparedStatementName)
+	if err != nil {
+		logrus.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorGeneral).
+			WithError(err).Errorln("PreparedStatement not present in registry")
+		return nil, false, ErrStatementNotPresentInRegistry
+	}
+
+	querySetting, ok := e.proxyHandler.registry.QuerySettingsByID(preparedStatementName)
+	if ok {
+		logrus.Debugln("Set query settings from registry")
+		e.querySelectSettings = querySetting
+	}
+
+	return base.NewOnQueryObjectFromStatement(executeQuery, e.parser), true, nil
+}
+
+func (e *PreparedStatementsQuery) onDeallocate(ctx context.Context, deallocateQuery *sqlparser.DeallocatePrepare) (base.OnQueryObject, bool, error) {
+	logrus.Debugln("PreparedStatementsQuery.Deallocate")
+
+	var preparedStatementName = deallocateQuery.PreparedStatementName.String()
+
+	if _, err := e.proxyHandler.registry.StatementByID(deallocateQuery.PreparedStatementName.String()); err != nil {
+		logrus.WithField(logging.FieldKeyEventCode, logging.EventCodeErrorGeneral).
+			WithError(err).Errorln("PreparedStatement not present in registry")
+		return nil, false, ErrStatementNotPresentInRegistry
+	}
+
+	e.proxyHandler.registry.DeleteStatementByID(preparedStatementName)
```

Comment:
let's clear `e.querySelectSettings` and `registry.QuerySettings` to avoid problems? remember we had some problems with not cleared `e.querySelectSettings` in some of OnQuery handler